velocity-custom-parser-example/pom.xml

```diff
@@ -0,0 +1,267 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.velocity</groupId>
+        <artifactId>velocity-engine-parent</artifactId>
+        <version>2.2-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>velocity-custom-parser-example</artifactId>
+    <name>Apache Velocity Custom Parser - Example</name>
+    <description>Custom Parser Example for Apache Velocity</description>
+
+    <packaging>jar</packaging>
+
+    <!--
+      This plugin demonstrates how to build a custom Velocity parser.
+      The proposed custom parser replaces '#' with '@' and '@' with '%',
+      so that it's suitable to use with Markdown template files, for instance.
+
+      The generated parser class is org.apache.velocity.runtime.parser.${parser.name}Parser,
+      and must be specified at runtime using the Velocity property parser.class :
+        parser.class = org.apache.velocity.runtime.parser.MyCustomParser
+    -->
+
+    <properties>
+        <!-- whether to display debug logs while parsing -->
+        <parser.debug>false</parser.debug>
+        <!-- parser name -->
+        <parser.name>Custom</parser.name>
+        <!-- character to substitute to '*' -->
+        <parser.char.asterisk>*</parser.char.asterisk>
+        <!-- character to substitute to '@' -->
+        <parser.char.at>%</parser.char.at>
+        <!-- character to substitute to '$' -->
+        <parser.char.dollar>$</parser.char.dollar>
+        <!-- character to substitute to '#' -->
+        <parser.char.hash>@</parser.char.hash>
+    </properties>
```

Comment:
The above text says that you only replace a subset of those chars -- yet you provide all of them. This bascially means that you cannot override them, but you *define* them. This should be properly noted.